nomenclature/processor/data_validator.py

```diff
@@ -86,9 +86,49 @@ def criteria(self):
         )
 
 
+class DataValidationCriteriaRange(DataValidationCriteria):
+    range: list[float] = Field(..., min_length=2, max_length=2)
+
+    @model_validator(mode="after")
+    def check_range_is_valid(self):
+        if self.range[0] > self.range[1]:
+            raise ValueError("Validation range is invalid: " + str(self.criteria))
```

Comment:
This error message does not contain enough information for the user to fix the issue:
```suggestion
            raise ValueError("Validation range must be ordered with the smaller value first: " + str(self.criteria))
```